Source/ARTPushActivationStateMachine.m

```diff
@@ -379,6 +381,19 @@ - (void)callUpdateFailedCallback:(nullable ARTErrorInfo *)error {
     #endif
 }
 
+- (void)registerForAPNS {
+#if !TARGET_OS_SIMULATOR
+    if ([NSThread isMainThread]) {
+        [[UIApplication sharedApplication] registerForRemoteNotifications];
+    }
+    else {
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [[UIApplication sharedApplication] registerForRemoteNotifications];
```

Comment:
If I was being really picky then I could point out that you could declare this block once and invoke it from both call sites to reduce repetition, but it's not like this is an overly complex method implementation so that's perhaps more of a tasting note. 😁

Reply:
Hmm, I'm not sure I've understood, `registerForRemoteNotifications` called twice - one time directly, and the other in the `dispatch_async` block. Or you mean to transform `dispatch_async(dispatch_get_main_queue(), ^{ }` into `dispatch_async_main(^{ }` ?

Reply:
The `[[UIApplication sharedApplication] registerForRemoteNotifications];` call in in two places, both of which can be encapsulated as a `dispatch_block_t` earlier on. Then you can either call it directly or pass it to `dispatch_async`.

Reply:
I got your point - calling `registerForRemoteNotifications` twice near to each other looks a bit weird. But wrapping this one-liner into another one (two liner actually) would be unnecessary overhead I think. What I would do, it's to remove `isMainThread` check entirely and call it always on `dispatch_get_main_queue` because `didRegisterForRemoteNotificationsWithDeviceToken` callback is asynchronous anyways. WDYT @QuintinWillison @lukasz-szyszkowski  @lawrence-forooghian

Reply:
Sounds good to me. It was just the DRY element I was commenting on - so never a problem to replace one line with a few lines, if the goal is reduction of repetition. 🤷

Reply:
https://github.com/ably/ably-cocoa/pull/1293/commits/baa71f75040b97f69dccbd244dfa69428c6c11b0